cfme/tests/infrastructure/test_vm_power_control.py

```diff
@@ -58,6 +60,59 @@ def if_scvmm_refresh_provider(provider):
         provider.refresh_provider_relationships()
 
 
+def check_power_options_when_on(soft_assert, vm, from_details):
+    soft_assert(
+        vm.is_pwr_option_available_in_cfme(option=vm.POWER_OFF, from_details=from_details))
+    soft_assert(
+        not vm.is_pwr_option_available_in_cfme(
+            option=vm.POWER_ON, from_details=from_details))
+    soft_assert(
+        vm.is_pwr_option_available_in_cfme(option=vm.SUSPEND, from_details=from_details))
+
+    # limited options for SCVMM providers
+    if vm.provider.type != "scvmm":
+
+        # RHEV VMs are slightly different in terms of options
+        if vm.provider.type == "rhevm":
+            soft_assert(
+                not vm.is_pwr_option_available_in_cfme(
+                    option=vm.GUEST_RESTART, from_details=from_details))
+            soft_assert(
+                not vm.is_pwr_option_available_in_cfme(
+                    option=vm.RESET, from_details=from_details))
+        else:
+            soft_assert(
+                vm.is_pwr_option_available_in_cfme(
+                    option=vm.GUEST_RESTART, from_details=from_details))
+            soft_assert(
+                vm.is_pwr_option_available_in_cfme(
+                    option=vm.RESET, from_details=from_details))
+
+        vm.is_pwr_option_available_in_cfme(
+            option=vm.GUEST_SHUTDOWN, from_details=from_details)
+    else:
+        pass
+
+
+def check_power_options_when_off(soft_assert, vm, from_details):
```

Comment:
You could even have it as a tuple that stores the operation (not, or not not) as well so that you can work with both.